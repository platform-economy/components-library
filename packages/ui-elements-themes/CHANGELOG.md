# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
