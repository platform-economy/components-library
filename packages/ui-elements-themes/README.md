# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
