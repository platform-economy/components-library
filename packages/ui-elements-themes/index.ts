export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
