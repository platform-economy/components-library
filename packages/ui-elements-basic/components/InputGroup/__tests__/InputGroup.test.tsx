import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import { InputGroup } from '../index';

import { TestTheme } from '@relayr/ui-elements-themes/';

const withTheme = (Component: React.ReactChild) => (
  <ThemeProvider theme={TestTheme}>
    {Component}
  </ThemeProvider>
);

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('InputGroup', () => {
      it('should render with default state', () => {
        const tree = TestRenderer
          .create(withTheme(
            <InputGroup />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render with mocked components inside', () => {
        const tree = TestRenderer
          .create(withTheme(
            <InputGroup>
              <input type="text"/>
              <button>Button</button>
            </InputGroup>))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
