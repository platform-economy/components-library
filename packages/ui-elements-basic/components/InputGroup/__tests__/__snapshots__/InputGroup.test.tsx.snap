// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components InputGroup should render with default state 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 35px;
  background-color: transparent;
}

.c0 * {
  max-height: 100%;
}

<div
  className="InputGroup c0"
/>
`;

exports[`ui-elements-basic Components InputGroup should render with mocked components inside 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 35px;
  background-color: transparent;
}

.c0 * {
  max-height: 100%;
}

<div
  className="InputGroup c0"
>
  <input
    type="text"
  />
  <button>
    Button
  </button>
</div>
`;
